relay-server/src/services/processor.rs

```diff
@@ -2627,7 +2627,14 @@ impl Service for EnvelopeProcessorService {
     type Interface = EnvelopeProcessor;
 
     fn spawn_handler(self, mut rx: relay_system::Receiver<Self::Interface>) {
-        let thread_count = self.inner.config.cpu_concurrency();
+        // Adjust thread count for small cpu counts to not have too many idle cores
+        // and distribute workload better.
+        let thread_count = match self.inner.config.cpu_concurrency() {
+            0 | 1 => 1,
+            2 | 3 => 2,
+            4 => 3,
+            conc => conc - 2,
```

Comment:
Have you run load tests with this? It would be good to know the actual needs here, if this is easy to do.

Reply:
No, this was simply a best effort guess for small amounts of cores going off our discussed `-2` starting point. 

Having a small amount of cores would reserve a much bigger % to the runtime than if we have a large amount of cores. E.g. for 2 cores allocating 1 thread to the processor would allocate 50% to the runtime, while having 12 cores only allocates 17% to the runtime.